2000-10-17  Jakub Jelinek  <jakub@redhat.com>

	* config/alpha/alpha.h (ASM_OUTPUT_MI_THUNK): Reload gp if needed.

--- gcc/config/alpha/alpha.h.jj	Mon Oct  9 11:18:54 2000
+++ gcc/config/alpha/alpha.h	Tue Oct 17 14:39:33 2000
@@ -2240,6 +2240,9 @@ do {									\
   const char *fn_name = XSTR (XEXP (DECL_RTL (FUNCTION), 0), 0);	\
   int reg;								\
 									\
+  if (! TARGET_OPEN_VMS && ! TARGET_WINDOWS_NT)				\
+    fprintf (FILE, "\tldgp $29,0($27)\n");				\
+									\
   /* Mark end of prologue.  */						\
   output_end_prologue (FILE);						\
 									\
